### DISCUSSION TITLE
Supabase: filter rows based on if any of their child relations has property

### DISCUSSION THREAD

**Discussion by kishansripada** [supabase/supabase#7647] [Questions] [answered]
In supabase I have a list of tracks and each track can have multiple artists (kept in a separate table). I only want tracks in which at least one of the artists associated with that track contains a certain property, we'll say their genre has to equal rap. 

```
  tracksWithAtLeast1ArtistWhoseGenreIsRap = await supabase
      .from("tracks")
      .select("name, artists ( genre ), id")
      .filter("artists.genre", "eq", "rap")

```
here is what i'm workin right right now but the issue is that I think this filters the selected artists within each track by genre rather than filtering the tracks based on if any of its artists has the genre "rap".

**Comment by steve-chavez** [comment, marked as answer]
Try with:

```js
  tracksWithAtLeast1ArtistWhoseGenreIsRap = await supabase
      .from("tracks")
      .select("name, artists!inner( genre ), id")
      .filter("artists.genre", "eq", "rap")
```

Check https://supabase.com/docs/reference/javascript/select#filtering-with-inner-joins.

**Reply by kishansripada** [reply]
worked beautifully!
so that i can understand, what makes this select tracks with AT LEAST one artist with the genre of rap as opposed to requiring that all artists have a genre of rap?

**Reply by steve-chavez** [reply]
Yes, basically it ensures `artists` **exist** for that particular query over `tracks`.
